aries_cloudagent/protocols/didexchange/v1_0/routes.py

```diff
@@ -362,6 +372,38 @@ async def didx_accept_request(request: web.BaseRequest):
     return web.json_response(result)
 
 
+@docs(
+    tags=["did-exchange"],
+    summary="Abandon or reject a DID Exchange",
+)
+@match_info_schema(DIDXConnIdMatchInfoSchema())
+@request_schema(DIDXRejectRequestSchema())
+@response_schema(ConnRecordSchema(), 200, description="")
+async def didx_reject(request: web.BaseRequest):
+    """Abandon or reject a DID Exchange."""
+    context: AdminRequestContext = request["context"]
+    outbound_handler = request["outbound_message_router"]
+
+    connection_id = request.match_info["conn_id"]
+    body = await request.json()
+    reason = body.get("reason")
+
+    profile = context.profile
+    didx_mgr = DIDXManager(profile)
+    try:
+        async with profile.session() as session:
+            conn_rec = await ConnRecord.retrieve_by_id(session, connection_id)
+        report = await didx_mgr.reject(conn_rec=conn_rec, reason=reason)
+    except StorageNotFoundError as err:
+        raise web.HTTPNotFound(reason=err.roll_up) from err
+    except (StorageError, WalletError, DIDXManagerError, BaseModelError) as err:
+        raise web.HTTPBadRequest(reason=err.roll_up) from err
+
+    await outbound_handler(report, connection_id=conn_rec.connection_id)
+    result = conn_rec.serialize()
```

Comment:
Should we return the updated connection record with the abandoned state? I would wonder why my returned connection didn't indicate that it was abandoned...

Reply:
Been a minute but I believe the `reject` call in the DID Exchange manager should modify the `conn_rec` so it should include the abandoned state.

Reply:
I'll do a quick double check on this but will merge if it returns an abandoned state as expected. Thanks for the review!

Reply:
Okay, confirmed that the state "abandoned" is returned.